src/other/index.jl

```diff
@@ -226,6 +226,10 @@ end
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
 @inline Base.getindex(x::AbstractIndex, notidx::Not) =
     setdiff(1:length(x), getindex(x, notidx.skip))
+@inline Base.getindex(x::AbstractIndex, notidx::Not{<:AbstractVector}) =
+    setdiff(1:length(x), getindex(x, unique(notidx.skip)))
+@inline Base.getindex(x::AbstractIndex, notidx::Not{InvertedIndices.NotMultiIndex}) =
```

Comment:
Why do we need to support this nested type?

Reply:
You mean why `NotMultiIndex` is needed? We need it because `Not(r"a", r"b")` is a valid selector and is not the same as `Not([r"a", r"b"])` (which we would get if we wrapped multiple positional arguments in a vector - like it is done for the case of integers).

The second issue is that InvertedIndices.jl is unaware of `Cols` which is defined in DataAPI.jl.

Reply:
I mean why isn't this just:
```suggestion
@inline Base.getindex(x::AbstractIndex, notidx::InvertedIndices.NotMultiIndex) =
```

Reply:
Ah OK I forgot this was how we handle multi indices.